relay-common/src/utils.rs

```diff
@@ -1,11 +1,49 @@
 use std::fmt;
+use std::mem;
 use std::str;
+use std::sync::{Arc, Mutex};
 
 use once_cell::sync::OnceCell;
 use regex::Regex;
 
 use crate::macros::impl_str_serde;
 
+/// Wraps either the provided pattern string, which will be consumed and replaced with compiled
+/// regex, when the `compiled` function is called.
+#[derive(Debug, Clone)]
+struct GlobRegex {
```

Comment:
I don't really like this name. I don't have good suggestions that I like, maybe something like `LazyPattern`?

Reply:
Not relevant anymore